### ISSUE TITLE
Incorrect response returned for unauthenticated `/works/merge` `GET` requests

### ISSUE THREAD

**Issue by Spaarsh** [internetarchive/openlibrary#10224]
### Problem

The issue arose when I was trying to triage a merge related issue. When I attempted to directly open the URL https://openlibrary.org/works/merge?records=OL15941796W,OL15400921W I only saw a blank page. 

Upon opening the console, I came to know that the server had returned a 403 response.

![image](https://github.com/user-attachments/assets/83dc6b7b-5ce5-4b51-a67f-b16a0b2b5943)

### Reproducing the bug

1. Go to https://openlibrary.org/works/merge?records=OL15941796W,OL15400921W
2. Do not login, if you have already, logout first and then try to go to the URL.

* Expected behavior:
* Actual behavior:

### Context

- Browser (Chrome, Safari, Firefox, etc): Firefox
- OS (Windows, Mac, etc):
- Logged in (Y/N): N
- Environment (prod, dev, local): prod

### Breakdown

I know there is a very low probability that any user comes across such a scenario, but it may be better if we respond with a basic page showing the 403 response. It would help contributors as well.

The general template for 404 response can be used for the 403 response as well. Or perhaps for any response other than a successful response.

Proposed solution:
![image](https://github.com/user-attachments/assets/44e75d1c-16be-41fc-b650-f7012ff04f2c)

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by Spaarsh** [comment]
I am willing to work on this issue.

**Comment by jimchamp** [comment]
Unless you found the link the that merge page while browsing Open Library, this is _very_ low priority.

Updating the [`/works/merge` `GET` endpoint](https://github.com/internetarchive/openlibrary/blob/2335e8cf9254a108ef236e12eaa0392b5f626393/openlibrary/plugins/upstream/code.py#L139-L154) to return `web.unauthorized()` when the patron should not have access will fix this issue.

**Comment by jimchamp** [comment]
@Spaarsh, can you provide an update on this?  Do you have any blockers?

**Comment by Spaarsh** [comment]
@jimchamp No, I am not facing any blocker. Sincere apologies for the delay. The current error raises a ["403 forbidden"](https://github.com/internetarchive/openlibrary/blob/be77d3d2b2ca5a1a179457e7c1851499619ee358/openlibrary/plugins/upstream/code.py#L136C1-L155C1), should I raise a web.forbidden()?
